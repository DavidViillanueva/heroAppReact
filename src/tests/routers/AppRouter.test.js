import { mount } from 'enzyme';
import React from 'react';
import { AuthContext } from '../../auth/AuthContext';
import AppRouter from '../../routers/AppRouter';

describe('<AppRouter /> test', () => {

    test('should render the login view if are not authenticated', () => {

        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: false
            }
        };

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        
    });

    test('should render MarvelView component if are authenticated', () => {
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'David'
            }
        };

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper.find('.navbar').exists() ).toBe( true );

    })
    
    
    
})
