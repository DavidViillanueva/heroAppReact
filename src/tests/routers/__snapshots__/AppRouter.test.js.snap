// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRouter /> test should render the login view if are not authenticated 1`] = `
<div>
  <div>
    <div
      className="container mt-5"
    >
      <h1>
        Login
      </h1>
      <hr />
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
